sklearn/impute.py

```diff
@@ -143,44 +136,34 @@ def fit(self, X, y=None):
                              " got strategy={1}".format(allowed_strategies,
                                                         self.strategy))
 
-        if self.axis not in [0, 1]:
-            raise ValueError("Can only impute missing values on axis 0 and 1, "
-                             " got axis={0}".format(self.axis))
-
         # Since two different arrays can be provided in fit(X) and
         # transform(X), the imputation data will be computed in transform()
-        # when the imputation is done per sample (i.e., when axis=1).
-        if self.axis == 0:
-            X = check_array(X, accept_sparse='csc', dtype=np.float64,
-                            force_all_finite=False)
-
-            if sparse.issparse(X):
-                self.statistics_ = self._sparse_fit(X,
-                                                    self.strategy,
-                                                    self.missing_values,
-                                                    self.axis)
-            else:
-                self.statistics_ = self._dense_fit(X,
-                                                   self.strategy,
-                                                   self.missing_values,
-                                                   self.axis)
+        # when the imputation is done per sample.
+        X = check_array(X, accept_sparse='csc', dtype=np.float64,
+                        force_all_finite=False)
+
+        if sparse.issparse(X):
+            self.statistics_ = self._sparse_fit(X,
+                                                self.strategy,
+                                                self.missing_values)
+        else:
+            self.statistics_ = self._dense_fit(X,
+                                               self.strategy,
+                                               self.missing_values)
 
         return self
 
-    def _sparse_fit(self, X, strategy, missing_values, axis):
+    def _sparse_fit(self, X, strategy, missing_values):
         """Fit the transformer on sparse data."""
         # Imputation is done "by column", so if we want to do it
         # by row we only need to convert the matrix to csr format.
-        if axis == 1:
-            X = X.tocsr()
-        else:
-            X = X.tocsc()
+        X = X.tocsc()
 
         # Count the zeros
         if missing_values == 0:
-            n_zeros_axis = np.zeros(X.shape[not axis], dtype=int)
```

Comment:
`not axis` should mean `1` if I am not wrong. By the way `not axis` is really not intuitive use here.

Reply:
OK! @glemaitre changed